### DISCUSSION TITLE
How do I add a calendar to the time axis?

### DISCUSSION THREAD

**Discussion by aaschwanden** [xCDAT/xcdat#381] [2. Questions]
I'm trying to calculate annual means for monthly data imported from a Pandas data frame, but I am receiving an error 

`KeyError: "This dataset's time coordinates do not have a 'calendar' encoding attribute set, which might indicate that the time coordinates were not decoded to datetime objects. Ensure that the time coordinates are decoded before performing temporal averaging operations."`

when I run the example code below:

```
df = pd.DataFrame({"time": pd.date_range("1992-1-1", periods=12, freq="M"), "data": range(0, 12)})
df.set_index("time", inplace=True, drop=True)
df_xr = df.to_xarray()
df_xr.temporal.group_average("data", freq="year", weighted=True)
```

How do I set the calendar so that the averaging works?

**Comment by pochedls** [comment]
If you have the original xarray dataset open, you should be able to retrieve the encoding with `encoding = ds.time.encoding.copy()` or manually set it with `encoding = {'calendar': 'standard'}`. You can then set this encoding to the time coordinate / dataset:

```
time = df_xr.time
time.encoding = encoding
df_xr['time'] = time
df_xr.temporal.group_average("data", freq="year", weighted=True)
```

I think this will lead to another error (xcdat needs the time bounds to compute the weighted annual average): 

> KeyError: "The coordinate variable 'time' has no 'bounds' attr. Set the 'bounds' attr to the name of the bounds data variable."

Could you instead subset the year 1992 as follows:

```
import xcdat as xc
ds = xc.open_dataset(fn)
dssub = ds.sel(time=slice("1992-01-01", "1992-12-31"))
dsa = dssub.temporal.group_average("data", freq="year", weighted=True)
```
